lua/rocks/operations/helpers.lua

```diff
@@ -29,11 +29,111 @@ local helpers = {}
 
 helpers.semaphore = nio.control.semaphore(1)
 
+---@class MultiMutRocksTomlWrapper
+---@field cache table<string, MultiMutRocksTomlWrapper> Cache for nested metatables
+---@field tables MutRocksTomlRef[] Rocks toml config metatables
+---@field paths? string[] File paths associated the rocks toml configs
+local MultiMutRocksTomlWrapper = {}
+MultiMutRocksTomlWrapper.__index = function(self, key)
+    -- Give preference to class methods/fields
+    if MultiMutRocksTomlWrapper[key] then
+        return MultiMutRocksTomlWrapper[key]
+    end
+    -- Find the key within the config tables
+    local tables = {}
+    for _, tbl in ipairs(self.tables) do
+        if tbl[key] ~= nil then
+            if type(tbl[key]) == "table" then
+                table.insert(tables, tbl[key])
+            else
+                return tbl[key]
+            end
+        end
+    end
+    -- If the value is a table, setup a nested metatable that uses the
+    -- inner tables of the config tables
+    if #tables > 0 then
+        if not self.cache[key] then
+            self.cache[key] = MultiMutRocksTomlWrapper.new(true, tables)
+        end
+        return self.cache[key]
+    end
+    return nil
+end
+MultiMutRocksTomlWrapper.__newindex = function(self, key, value)
+    for _, tbl in ipairs(self.tables) do
+        if tbl[key] ~= nil then
+            tbl[key] = value
+            return
+        end
+    end
+    -- If key not found in any table, add it to the first table
+    self.tables[1][key] = value
+end
+
+--- Write to all rocks toml config files
```

Comment:
suggestion: we should hint that this function needs to be called in an async context for the write to complete before it returns.

```suggestion
---@type async fun()
--- Write to all rocks toml config files
```